pkg/executor/sortexec/parallel_sort_spill_test.go

```diff
@@ -143,3 +144,16 @@ func TestParallelSortSpillDiskFailpoint(t *testing.T) {
 		failpointDataInMemoryThenSpillTest(t, ctx, exe, sortCase, dataSource)
 	}
 }
+
+func TestIssue55344(t *testing.T) {
+	store := testkit.CreateMockStore(t)
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("use test")
+	tk.MustExec("drop table if exists t;")
+	tk.MustExec("CREATE TABLE t0(c0 BOOL);")
+	tk.MustExec("INSERT INTO mysql.opt_rule_blacklist VALUES('predicate_push_down'),('column_prune');")
+	tk.MustExec("ADMIN reload opt_rule_blacklist;")
```

Comment:
I think the `opt_rule_blacklist` should be reset back after the tests

Reply:
> I think the `opt_rule_blacklist` should be reset back after the tests

okk